nss_role.c

```diff
@@ -38,6 +39,12 @@ enum nss_status _nss_role_initgroups_dyn(char *user, gid_t main_group,
         goto libnss_role_out;
     }
 
+    result = librole_get_directory_files(LIBROLE_CONFIG_DIR, &G);
+    if (LIBROLE_OK != result) {
```

Comment:
As far as I understand `librole_get_directory_files` returns an error if `/etc/roles.d` does not exist. Thus this function will bail out in a system without `/etc/roles.d` directory. This violates the principle of least surprise and makes upgrades more difficult than necessary.